helpers/browser/driver.js

```diff
@@ -74,48 +67,92 @@ class ChromeProtocol {
     }, 15 * 1000);
   }
 
-  subscribeToServiceWorkerDetails(cb, resolve) {
-    var chrome = this.instance();
+  subscribeToServiceWorkerDetails(fn) {
+    var chrome = this._instance;
+
+    return new Promise(function(res, rej) {
+      chrome.ServiceWorker.enable();
+      // chrome.on("ServiceWorker.workerCreated", log)
+      // chrome.on("ServiceWorker.workerRegistrationUpdated", log)
+      chrome.on("ServiceWorker.workerVersionUpdated", data => {
+        res(fn(data));
+      });
+    });
+  }
+
+  static getEvaluationContextFor(url, instance) {
+    return new Promise((res, rej) => {
+      var errorTimeout = setTimeout((_ => rej(new Error(`No Evaluation context found for ${url}`))), 4000);
 
-    chrome.ServiceWorker.enable();
-    // chrome.on("ServiceWorker.workerCreated", log)
-    // chrome.on("ServiceWorker.workerRegistrationUpdated", log)
-    chrome.on("ServiceWorker.workerVersionUpdated", data => {
-      cb(data, resolve);
+      instance.on('Runtime.executionContextCreated', evalContext => {
+        if (evalContext.context.origin.indexOf(url) !== -1) {
```

Comment:
multiple contexts can be created so i actually expect this to fail for all but 1 case :)